torchok/data/datasets/classification/classification.py

```diff
@@ -93,14 +120,19 @@ def __init__(self,
         self.target_dtype = target_dtype
         self.multilabel = multilabel
         self.lazy_init = lazy_init
-        self.csv_path = csv_path
+        self.annotation_path = annotation_path
 
-        csv_path = self.data_folder / self.csv_path
         dtype = {self.input_column: 'str', self.target_column: 'str' if self.multilabel else 'int'}
 
-        self.csv = pd.read_csv(csv_path, dtype=dtype)
+        if annotation_path.endswith('.csv'):
+            self.df = pd.read_csv(self.data_folder / annotation_path, dtype=dtype)
+        elif annotation_path.endswith('.pkl'):
+            self.df = pd.read_pickle(self.data_folder / annotation_path)
+        else:
+            raise ValueError('Detection dataset error. Annotation path is not in `csv` or `pkl` format')
+
         if not self.lazy_init and not self.test_mode:
-            self.csv[self.target_column] = self.csv[self.target_column].apply(self.process_function)
+            self.df[self.target_column] = self.df[self.target_column].apply(self.process_function)
```

Comment:
Also this change force you to change each dataset that we have

Reply:
Add warning for that